rebase_v1.13.1_report.md

```diff
@@ -0,0 +1,38 @@
+- `origin/main` is **270 commits** ahead of `v1.11.3` and **501 commits** behind `v1.13.1`; there are 82 merge commits and 123 commits containing `UPSTREAM` markers that will need manual triage.
+- Numerous cherry-picked upstream fixes (e.g., PRs #6692, #6836, #7083) already exist in the target release, implying they must be dropped or squashed during the rebase.
+
+## Carry Patches and OpenShift Packaging
+- Long-running OpenShift carry chain touches `.ci-operator.yaml`, `Dockerfile.ocp`, `Dockerfile.openshift`, `Makefile` overrides (`test`, `dep-ensure`), and repeated image metadata synchronizations with ART. These files do not exist upstream; they must be reintroduced cleanly after the rebase.
+- The external plugin `ocp_dnsnameresolver` is currently vendored along with large slices of `github.com/openshift/{api,client-go,...}`. Upstream `v1.13.1` has no `vendor/` tree, so we must decide between re-vendoring the entire OpenShift dependency set or switching to module-aware builds that can fetch these deps.
+- Several carry commits revert or modify upstream behavior (e.g., wildcard warning toggles, cache CD-bit handling, bufsize defaults). Each needs reassessment against upstream changes to avoid reintroducing bugs that have since been fixed differently.
+
+## Toolchain and Dependency Shifts
+- Upstream now enforces Go **1.24+** (release notes call out 1.25.2) via `.go-version` and `GOTOOLCHAIN`. Our builders and CI images must ship that compiler, otherwise `go` will attempt to auto-download toolchains (blocked in disconnected builds).
+- `go.mod` migrates from legacy dependencies (`aws-sdk-go` v1, DataDog v1, etc.) to new major versions (`aws-sdk-go-v2`, `dd-trace-go/v2`, OpenTelemetry packages, etc.), and bumps Kubernetes libs to **v0.34.1**. These introduce breaking API changes that our OpenShift carry code (including the external plugin) must accommodate.
+- Removal of the `vendor/` directory in upstream implies all historical vendoring scripts (`dep-ensure`, Dockerfiles that copied `vendor/`) and downstream assumptions about offline builds must be rewritten.
+
+## Core Runtime and Plugin Differences
+- `plugin.cfg` in `v1.13.1` adds new built-ins (`quic`, `multisocket`, `nomad`) and exposes `COREDNS_PLUGINS` overrides. We must reinsert `ocp_dnsnameresolver` without breaking the new ordering constraints or the generated `zplugin.go` / `zdirectives.go`.
+- Core server code (`core/dnsserver/*`, `coremain/run.go`) gained QUIC/HTTPS refactors, new configuration paths, and extensive tests. OpenShift-specific patches touching these surfaces (e.g., cache behavior, bufsize handling) will likely conflict and must be forward-ported carefully.
+- The Kubernetes plugin evolved across several releases alongside controller-runtime API changes. Custom patches we carry for OpenShift bugs (OCPBUGS-15755, -27397, -34619, etc.) need a fresh review because upstream code may have diverged significantly.
```

Comment:
Are these really "&lt;carry&gt;" patches?  If they are backports of upstream fixes that we get with the bump to CoreDNS v1.13.1, then they can be dropped.

Reply:
Updated the Kubernetes-plugin note to reflect that our OCP bug fixes were upstream backports already merged in CoreDNS v1.13.1, so those commits will drop automatically; the real follow-up is ensuring ocp_dnsnameresolver still builds against the newer k8s.io libraries.